runbot/common.py

```diff
@@ -22,6 +22,21 @@
 dest_reg = re.compile(r'^\d{5,}-.+$')
 
 
+def transactioncache(method):
+    def wrapper(self, *args, **kwargs):
+        assert not self.ids
+        cache = self.env.cr.cache
+        method_key = (self._name, method.__name__)
+        params_key = (args, frozenset(kwargs.items()))
+        # should check id key is serializable
+        if method_key not in cache:
+            cache[method_key] = {}
+        if params_key not in cache[method_key]:
+            cache[method_key][params_key] = method(self.with_context({}), *args, **kwargs)
+        return cache[method_key][params_key]
+    wrapper.clear_transaction_cache = lambda self: self.env.cr.cache.pop((self._name, method.__name__), None)
```

Comment:
Could you not use the method itself as a cache key?
```suggestion
        method_key = (method, )
        params_key = (args, frozenset(kwargs.items()))
        # should check id key is serializable
        if method_key not in cache:
            cache[method_key] = {}
        if params_key not in cache[method_key]:
            cache[method_key][params_key] = method(self.with_context({}), *args, **kwargs)
        return cache[method_key][params_key]
    wrapper.clear_transaction_cache = lambda self: self.env.cr.cache.pop((method, ))
```